src/render.rs

```diff
@@ -191,16 +189,20 @@ impl Renderer {
         cursor_surf.damage_buffer(0, 0, w as _, h as _);
         cursor_surf.commit();
         self.cursor_surf = Some(cursor_surf);
+
+	return Ok(());
     }
 
     pub fn set_cursor(&mut self, wl: &WaylandClient, mouse: &WlPointer, serial: u32) {
-        if self.cursor_surf.is_none() {
-            self.setup_cursor(wl);
-        }
-        if self.cursor_surf.is_some() {
-            let (x, y) = self.cursor_spot;
-            mouse.set_cursor(serial, self.cursor_surf.as_ref(), x, y);
-        }
+	match self.cursor_surf {
+	    Some(_) => {
+		let (x, y) = self.cursor_spot;
+		mouse.set_cursor(serial, self.cursor_surf.as_ref(), x, y);
+	    },
+	    None => {
+		_ = self.setup_cursor(wl);
```

Comment:
The first time this function is run, it will fail to set the cursor even if the cursor is found.  In fact, this function didn't need to change to address the error.